sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -264,8 +269,7 @@ case class CachedRDDBuilder(
   }
 
   private def buildBuffers(): RDD[CachedBatch] = {
-    val cb = if (cachedPlan.supportsColumnar &&
-        serializer.supportsColumnarInput(cachedPlan.output)) {
+    val cb = if (supportsColumnarInput) {
       serializer.convertColumnarBatchToCachedBatch(
```

Comment:
what's the default implementation for this? I'm wondering when users need to care about input is columnar or not.

Reply:
The vanilla Spark default implementation is `throw new IllegalStateException("Columnar input is not supported")`. So only the people who use a custom cache serializer will be affected.

Reply:
Is this information really useful? It's about the cache-building phase and has nothing to do with the following cache read performance.